magnum-kyo/src/test/scala/com/augustnagro/magnum/magkyo/PgKyoTests.scala

```diff
@@ -0,0 +1,64 @@
+package com.augustnagro.magnum.magkyo
+
+import com.augustnagro.magnum.*
+import com.dimafeng.testcontainers.PostgreSQLContainer
+import com.dimafeng.testcontainers.munit.fixtures.TestContainersFixtures
+import munit.{AnyFixture, FunSuite, Location}
+import org.postgresql.ds.PGSimpleDataSource
+import org.testcontainers.utility.DockerImageName
+import kyo.*
+
+import java.nio.file.{Files, Path}
+import scala.util.Using
+import scala.util.Using.Manager
+import javax.sql.DataSource
+
+class PgZioTests extends FunSuite, TestContainersFixtures:
+
+  immutableRepoKyoTests(this, PostgresDbType, xa)
+
+  val pgContainer = ForAllContainerFixture(
+    PostgreSQLContainer
+      .Def(dockerImageName = DockerImageName.parse("postgres:17.0"))
+      .createContainer()
+  )
+
+  override def munitFixtures: Seq[AnyFixture[?]] =
+    super.munitFixtures :+ pgContainer
+
+  def xa(): TransactorKyo =
```

Comment:
maybe use a `IO` here. Even if it's a test, it's better to do it the `right` way.
```scala
def xa: TransactorKyo < (IO & Abort[Throwable])
```